CMakeLists.txt

```diff
@@ -24,6 +24,7 @@ endif ()
 option(JSON_BuildTests "Build the unit tests when BUILD_TESTING is enabled." ON)
 option(JSON_Install "Install CMake targets during install step." ON)
 option(JSON_MultipleHeaders "Use non-amalgamated version of the library." OFF)
+option(JSON_ExplicitConversions "Enable explicit conversion operator" OFF)
```

Comment:
I'm not sure about the wording. Isn't the flag more about *disabling* implicit conversions? At least this would be the difference between the current code and when the flag is set.

Reply:
The technical side is making the conversion operator explicit.  If you're looking at the option from the POV of how it changes behavior, then yes it is disabling implicit conversions, but if you are looking at the effect the option has on the code, then making the conversion operator explicit makes sense.  (This assumes that this is all that this option does.  If it's doing more than that, then it may make sense to word it the other way.)

Reply:
The next question is what happens to this option in the next major version where we can make this the default?  Does it become "Enable implicit conversion operator"?

Reply:
Hm. To me, an explicit conversion seems less a feature than an implicit one. So I would rather expect a switch controlling whether or not I would like to have implicit conversions. So for the time being,

```cmake
option(JSON_ImplicitConversions "Enable implicit conversion operators" ON)
```

Reply:
@nlohmann The current behavior is broken, I wouldn't reason in terms of which one is more a feature, but rather as a new behavior

Reply:
Ah, I wasn't thinking of the possibility of having the "use CMake" option being different than the "just include the header" option.  That would make it so that we don't have to change the name of the CMake option in the next major version.

Reply:
I'm still struggling with the wording. To me, the switch "enable explicit conversion" sounds like an additional feature. Instead, it breaks existing code, because the assumed implicit conversions do not work any longer. If the code already only used explicit conversions, then flipping the switch does not change anything.